iam/cloud-client/snippets/modify_policy_remove_member.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http:#www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# [START iam_modify_policy_remove_member]
+from typing import Dict, List, Union
+
+
+def modify_policy_remove_member(
+    bindings: List[Dict[str, Union[str, List[str]]]], role: str, member: str
+) -> List[Dict[str, Union[str, List[str]]]]:
+    """
+    Remove user from policy binding.
+
+    bindings: Policy attached to the project, which have to be modified.
+    role: role to which member need to be added.
+    member: The principals requesting access.
+
+    Possible format for member:
+        * user:{emailid}
+        * serviceAccount:{emailid}
+        * group:{emailid}
+        * deleted:user:{emailid}?uid={uniqueid}
+        * deleted:serviceAccount:{emailid}?uid={uniqueid}
+        * deleted:group:{emailid}?uid={uniqueid}
+        * domain:{domain}
+    """
+    for bind in bindings:
```

Comment:
@m-strzelczyk I've done 2 snippets in a different way and would like to align, which approach will meet expectations:
- **iam_modify_policy_add_member** is rewritten using e2e approach requesting and modifying the policies
- **iam_modify_policy_remove_member** is rewritten in the same was it was done in apiary

Reply:
OK, thanks for providing the options.

I think the best format would be to have structure similar to the `add_member` version, but with a twist. Perhaps make it `iam_modify_policy_add_member(project_id, role, member)` and have it fetch the current policy and update it. Similar with remove member.

Also, would it be possible to add `etag` handling? In case of IAM policies it's quite important not to overwrite changes that others might be doing in parallel and etags are the way to do it. If you send a policy update with etag of the previous policy, it will be checked if the policy in the cloud didn't change. If it did, you will need to repeat the whole operation basically with the new base for your change.

Reply:
Makes sense, since the entire communication is supposed to be in a way _read->modify->write_. I've updated it accordingly.

Speaking of `etag` - that's a good point, I've missed that. I've added a comment for `set_policy`, with encourage to add backoff to be sure changes applied.

Regarding `etag` in tests - it is tricky, cause we want to restore policy to default no matter what. Because of that `etag` has to be omitted. As a consequence `etag` can't be tracked, since tests are running in parallel, hitting the same project =(